lib/api/controllers/documentController.js

```diff
@@ -146,9 +146,13 @@ class DocumentController extends BaseController {
    * @returns {Promise<Object>}
    */
   mGet (request) {
-    const
-      ids = this.getBodyArray(request, 'ids'),
-      { index, collection } = this.getIndexAndCollection(request);
+    let ids;
+    if (request.input.body) {
+      ids = this.getBodyArray(request, 'ids');
+    } else {
+      ids = request.input.args._ids;
```

Comment:
Don't trust values provided by users, and use `ids = this.getArray(request, 'ids');`

Also, why is it `_ids` here and `ids` if put in the body part of the document?

Otherwise, submitting the following request to the API will make Kuzzle return a TypeError to the requesting user:

`{"controller":"document", "action":"mGet", "_ids":true, "index": "foo", "collection": "bar"}`

Reply:
Also, why is it `_ids` here and `ids` if put in the body part of the document?